python_scripts/cross_validation_learning_curve.py

```diff
@@ -93,9 +93,8 @@ def make_cv_analysis(regressor, data, target):
 
 # %%
 import matplotlib.pyplot as plt
-import seaborn as sns
 
-sns.displot(scores_sample_sizes, kind="kde")
+scores_sample_sizes.plot.kde()
```

Comment:
I wonder if we should refrain from using kernel density estimation plots and instead use histograms.

- con for KDE plots: it is quite a complex concept, our audience does not necessarily what "kde" stands for and might find it too magic and therefore frustrating and we do not want to digress to explain what it is. 
- pro for using KDE plots: the figures is more beautiful (less crowded than an histogram)
- (minor) pro for using KDE plots: they come with automatically selected kernel size by default instead of having to tweak the number of bins to avoid artifacts (although 10 to 30 bins is usually safe)
- pro for using KDE plots: our not-completely-beginners readers might discover that it exists without too much cognitive overload and that it's a nice alternative to histograms and that adds value to our teaching.


Any opinion @GaelVaroquaux in particular?

Reply:
For this particular plot, we need the KDE. The histogram will not work. However, while teaching this part, I thought it would be almost unnecessary to have it in the course (maybe move it as an exercise instead). We are using the `sklearn.model_selection.learning_curve` in the subsequent section and this is good enough I think.

Reply:
I don't really follow. Can you please clarify what is not necessary? Manually creating the learning curves by hand with the `make_cv_analysis` and directly introducing the concepts with the `learning_curve` function directly?

+1 if so.

Reply:
> Manually creating the learning curves by hand with the make_cv_analysis and directly introducing the concepts with the learning_curve function directly?

Yes exactly. And optionally have `make_cv_analysis` as an exercise if it fits.